R/utilities-configuration-plan.R

```diff
@@ -214,3 +272,28 @@ tlfScale <- function(configurationScale) {
   # tolower is used to ensure that there is no issue with caps from field values
   ConfigurationScales[[tolower(configurationScale)]]
 }
+
+#' @title get
+#' @description Translate value of `Scaling` property from configuration plan into tlf `scale`
+#' @param pathArray Configuration plan path as an array for observed data
+#' @param configurationPlan configuration plan object
+#' @param simulation simulation object
+#' @return molecular weight in base unit or NA if not found
+getMolWeightForObservedData <- function(pathArray, configurationPlan, simulation) {
+  # pathArray is assumed to include the Observed data id as first value
+  molWeight <- configurationPlan$getMolWeightForObservedData(pathArray[1])
+  if (!is.na(molWeight)) {
+    return(molWeight)
+  }
+  # pathArray is assumed to include the compound name as before the last value
+  compoundName <- utils::head(utils::tail(pathArray, 2), 1)
+  # In the current version, getAllMoleculePathsIn is faster and lighter than getAllMoleculesMatching
+  # since only a path name for the molecule is necessary
+  allMoleculePaths <- ospsuite::getAllMoleculePathsIn(simulation)
+  pathForCompoundName <- utils::head(allMoleculePaths[grepl(compoundName, allMoleculePaths)], 1)
+  # When compound is not found in simulation, return NA

```

Comment:
not sure if it's the optimal way. 
Created an issue for that: #507 
For now we can take this implementation but we should check if there is a better way for doing this in the ospsuite